launcher/ui/widgets/ThemeCustomizationWidget.cpp

```diff
@@ -0,0 +1,149 @@
+/// ui->iconsLabel->setVisible(features& ThemeFields::ICONS);
+/// ui->widgetStyleComboBox->setVisible(features& ThemeFields::WIDGETS);
+/// ui->widgetThemeLabel->setVisible(features& ThemeFields::WIDGETS);
+/// ui->backgroundCatComboBox->setVisible(features& ThemeFields::CAT);
+/// ui->backgroundCatLabel->setVisible(features& ThemeFields::CAT);
+/// 
+/// original Method Two:
+///     if (!(features & ThemeFields::ICONS)) {
+///         ui->formLayout->setRowVisible(0, false);
+///     }
+///     if (!(features & ThemeFields::WIDGETS)) {
+///         ui->formLayout->setRowVisible(1, false);
+///     }
+///     if (!(features & ThemeFields::CAT)) {
+///         ui->formLayout->setRowVisible(2, false);
+///     }
+/// </summary>
+/// <param name="features"></param>
+void ThemeCustomizationWidget::showFeatures(ThemeFields features) {
+    ui->iconsComboBox->setEnabled(features & ThemeFields::ICONS);
+    ui->iconsLabel->setEnabled(features & ThemeFields::ICONS);
+    ui->widgetStyleComboBox->setEnabled(features & ThemeFields::WIDGETS);
+    ui->widgetThemeLabel->setEnabled(features & ThemeFields::WIDGETS);
+    ui->backgroundCatComboBox->setEnabled(features & ThemeFields::CAT);
+    ui->backgroundCatLabel->setEnabled(features & ThemeFields::CAT);
+}
+
+void ThemeCustomizationWidget::applyIconTheme(int index) {
+    auto settings = APPLICATION->settings();
+    auto originalIconTheme = settings->get("IconTheme").toString();
+    auto& newIconTheme = m_iconThemeOptions[index].first;
+    settings->set("IconTheme", newIconTheme);
+
+    if (originalIconTheme != newIconTheme) {
+        APPLICATION->applyCurrentlySelectedTheme();
+    }
+
+    emit currentIconThemeChanged(index);
+}
+
+void ThemeCustomizationWidget::applyWidgetTheme(int index) {
+    auto settings = APPLICATION->settings();
+    auto originalAppTheme = settings->get("ApplicationTheme").toString();
+    auto newAppTheme = ui->widgetStyleComboBox->currentData().toString();
+    if (originalAppTheme != newAppTheme) {
+        settings->set("ApplicationTheme", newAppTheme);
+        APPLICATION->applyCurrentlySelectedTheme();
+    }
+
+    emit currentWidgetThemeChanged(index);
+}
+
+void ThemeCustomizationWidget::applyCatTheme(int index) {
+    auto settings = APPLICATION->settings();
+    settings->set("BackgroundCat", m_catOptions[index].first);
+
+    emit currentCatChanged(index);
+}
+
+void ThemeCustomizationWidget::applySettings()
+{
+    applyIconTheme(ui->iconsComboBox->currentIndex());
+    applyWidgetTheme(ui->widgetStyleComboBox->currentIndex());
+    applyCatTheme(ui->backgroundCatComboBox->currentIndex());
+}
+void ThemeCustomizationWidget::loadSettings()
+{
+    auto settings = APPLICATION->settings();
+
+    auto iconTheme = settings->get("IconTheme").toString();
+    for (auto& iconThemeFromList : m_iconThemeOptions) {
+        ui->iconsComboBox->addItem(QIcon(QString(":/icons/%1/scalable/settings").arg(iconThemeFromList.first)), iconThemeFromList.second);
```

Comment:
oh, I guess I didn't look in the right spot, I added the icon stuff to the ComboBox after adding the cat, which was originally using such a resource string. That ofc doesn't make a lot of sense....

Reply:
ah, getThemedIcon gets the icon (string in parameter), from the currently active icon theme. I want a icon from a specific theme. So that call is not useful.
![image](https://user-images.githubusercontent.com/31988415/211570455-5f8c2e23-ef0b-4f7a-9eff-8e4649c3a77c.png)